### PR TITLE
Add support for router basename

### PR DESCRIPTION
**Pull request by djhi**
- [x] Add support for router basename prop
- [x] Documentation

### CODE REVIEW

File: docs/Admin.md [hunk 1]
```diff
@@ -470,7 +471,17 @@ const App = () => (
 );
 ```
 
-**Caution**: Do not use the 5.x version of the `history` package. It's currently incompatible with another dependency of react-admin, `connected-react-router`. `history@4.10.1` works fine. 
+## `basename`
+
+Use this prop to make all routes and links in your Admin relative to a "base" portion of the URL pathname that they all share. This is useful when including the Admin inside a larger app. See https://reactrouter.com/docs/en/v6/api#router for more informations.
```
Comment (WiXSL) [inline docs/Admin.md:476]: `informations.` => **information**.

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/Admin.md (hunk 1 above)
Comment (djhi) [reply docs/Admin.md:476]:
```suggestion
Use this prop to make all routes and links in your Admin relative to a "base" portion of the URL pathname that they all share. This is useful when including the Admin inside a larger app. See https://reactrouter.com/docs/en/v6/api#router for more information.
```

**Comment by djhi** [comment]
With or without the basename, the app is served the same way and works

**Comment by fzaninotto** [comment]
So what's the use of the basename prop?

**Comment by djhi** [comment]
Actually, it only matters when using the browser history and not serving the app from the domain root.

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Then you should change the doc accordingly, to explain that it's only necessary when using browser history (and modify the example to include the history).

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**